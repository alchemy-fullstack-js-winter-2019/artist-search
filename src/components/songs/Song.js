import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { getLyrics } from '../../services/getArtistsApi';
import SongLyrics from './SongLyric';

export default class Song extends Component {
  state = {
    songLyrics: ''
  };

  static propTypes = {
    workTitle: PropTypes.string.isRequired,
    artistName: PropTypes.string.isRequired
  };

  getSongLyrics = () => {
    console.log(this.props.artistName, 'hello');
    getLyrics(this.props.artistName, this.props.workTitle)
      .then(response => {
        console.log('response', response.songLyrics);
        this.setState({ songLyrics: response.songLyrics });
      });
  }

  render() {
    const { songLyrics } = this.state;
    return (
      <>
        <button onClick={this.getSongLyrics}> Get Lyrics </button>
        <SongLyrics songLyrics={songLyrics} />
      </>
    );
  }
}

