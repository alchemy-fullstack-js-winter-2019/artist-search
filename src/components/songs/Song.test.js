import React from 'react';
import renderer from 'react-test-renderer';
import Song from './Song';

describe('Song', () => {
  it('matches a snapshot', () => {
    const song = {
      title: 'em',
      language: 'us',
      disambiguation: 'rap'
    };
    const tree = renderer.create(
      <Song song={song} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
