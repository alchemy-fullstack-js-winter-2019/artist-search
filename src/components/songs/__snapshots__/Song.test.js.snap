// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Song matches a snapshot 1`] = `
Array [
  <h2>
    em
  </h2>,
  <p>
    rap
  </p>,
  <p>
    us
  </p>,
]
`;
